### ISSUE TITLE
the app some error

### ISSUE THREAD

**Issue by zjh02249** [cloud-annotations/object-detection-android#1]
E/flitecameradem: [qarth_debug:]  get PatchStore::createDisableExceptionQarthFile method fail.
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: android.example.com.tflitecamerademo, PID: 6675
    java.lang.IllegalArgumentException: Cannot copy between a TensorFlowLite tensor with shape [1, 1917, 1, 4] and a Java object with shape [1, 1917, 1, 2].
        at org.tensorflow.lite.Tensor.throwExceptionIfTypeIsIncompatible(Tensor.java:242)
        at org.tensorflow.lite.Tensor.copyTo(Tensor.java:116)
        at org.tensorflow.lite.NativeInterpreterWrapper.run(NativeInterpreterWrapper.java:157)
        at org.tensorflow.lite.Interpreter.runForMultipleInputsOutputs(Interpreter.java:250)
        at com.example.android.tflitecamerademo.ImageClassifier.classifyFrame(ImageClassifier.java:107)
        at com.example.android.tflitecamerademo.CustomCameraFragment.lambda$new$0$CustomCameraFragment(CustomCameraFragment.java:109)
        at com.example.android.tflitecamerademo.-$$Lambda$CustomCameraFragment$I_thQX2Vv8tCIzJNCQ28EJmbkMY.onClick(Unknown Source:2)
        at android.view.View.performClick(View.java:6663)
        at android.view.View.performClickInternal(View.java:6635)
        at android.view.View.access$3100(View.java:794)
        at android.view.View$PerformClick.run(View.java:26199)
        at android.os.Handler.handleCallback(Handler.java:907)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:216)
        at android.app.ActivityThread.main(ActivityThread.java:7593)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)

**Comment by bourdakos1** [comment]
Hmm I think there’s a bug on my end, I’ll look into this today

**Comment by Shashanksoni06** [comment]
I am having the same issue. please respond.

**Comment by xunaizhong20** [comment]
+1
I am having the same issue. please respond.

**Comment by bourdakos1** [comment]
I’m on vacation at the moment. If someone can solve this issue and make a pull request I’ll be glad to merge it

**Comment by Shashanksoni06** [comment]
how many labels are you using xunaizhong20 ..??

**Comment by Shashanksoni06** [comment]
Update: I tested it for quite a few models and figured out that models with label count greater than or equal to 3 are working seamlessly.

**Comment by bourdakos1** [comment]
That’s strange, I was hoping I accidentally hard coded the label count. I’m not sure why it fails with 1 or 2 labels 😕

**Comment by bourdakos1** [comment]
@shashanksoni06 if you hardcode `mLabelList.size()` when you have 2 labels are you able to get it to run?

**Comment by Shashanksoni06** [comment]
@bourdakos1 No, even hard coding `mLabelList.size()` to sizes other than 3 are not working. It crashes giving the above exception.

**Comment by xunaizhong20** [comment]
@bourdakos1 @Android-BRUCE @Shashanksoni06 
com.example.android.tflitecamerademo.ImageClassifier
line99
// The models output expects a shape of [1 x 1917 x 1 x labels_size + 1]
float[][][][] _boxPredictionsPointer_ = new float[1][NUM_ANCHORS][1][mLabelList.size() + 2];
![311559100431_ pic](https://user-images.githubusercontent.com/15344030/58527322-bcc4b980-8204-11e9-82e7-9594e12690be.jpg)

**Comment by Shashanksoni06** [comment]
@xunaizhong20  you got it running for any number of labels you want ??

**Comment by xunaizhong20** [comment]
@Shashanksoni06 Never tried

**Comment by Shashanksoni06** [comment]
@xunaizhong20 
It is working because you may be using 2 labels. that makes `mLabelList.size() + 2` equals to 4. If you put any other value the app will crash.

**Comment by xunaizhong20** [comment]
@Shashanksoni06 You're right～but you can if "mLabelList.size()"~Do you understand what I mean～

**Comment by bourdakos1** [comment]
Can someone confirm that the tensor expects a shape of max(4, mLabelList.size())?

**Comment by stspanho** [comment]
@bourdakos1 works for me (with 2 labels). Thx

**Comment by bourdakos1** [comment]
@stspanho thanks :) can someone try hardcoding 4 and using like 5 labels?

**Comment by bourdakos1** [comment]
Oh man I’m dumb.... it’s a typo
This:
```
float[][][][] _boxPredictionsPointer_ = new float[1][NUM_ANCHORS][1][mLabelList.size() + 1];
```
Should be this:
```
float[][][][] _boxPredictionsPointer_ = new float[1][NUM_ANCHORS][1][4];
```
It’s always 4, because it represents each of the box coordinates. I think I copy pasted the class line twice and tested with 3 labels 🤦‍♂️

If somebody wants to make this change and make a pull request I’ll merge it. Otherwise it will have to wait a few days for me to be at a computer.

**Comment by Shashanksoni06** [comment]
@bourdakos1 I see why it worked when i hard coded the second line too. LOL . thanks,it works like a charm now.

**Comment by bourdakos1** [comment]
fixed with #2, thanks everyone for all the help!
